### PR TITLE
testing sidecar pattern for jenkins pod template

### PR DESCRIPTION
**Pull request by jitendar-singh**
- Define the feature file for testing [JKNS-257](https://issues.redhat.com/browse/JKNS-257)

### CODE REVIEW

**Comment by jitendar-singh** [comment]
For the test would be using this [Jenkinsfile](https://github.com/akram/pipes/blob/container-nodes/Jenkinsfile)

File: smoke/features/sidecar-pattern-jenkins.feature
Comment (akram) [inline smoke/features/sidecar-pattern-jenkins.feature:22]: There isn't any point that we check that we are using a PodTemplate sidecar like patten.

Comment (jitendar-singh) [reply smoke/features/sidecar-pattern-jenkins.feature:22]: As agreed over slack, its not required now

**Comment by gabemontero** [comment]
remember @jitendar-singh ... you want someone to look at a PR of yours, use

/assign @gabemontero

**Comment by gabemontero** [comment]
also looks like we'll need a bz for merge, unless

@adambkaplan - how hard is it to change the jenkins repo to no Feature Freeze ?

**Review by gabemontero — CHANGES_REQUESTED** [review]
In addition to use the pre defined java side car pod template now included in the image, you should attempt to us the examples we added into docs

see https://github.com/openshift/openshift-docs/pull/39873 where it provides some sample config maps etc. which can be used

**Comment by gabemontero** [comment]
> also looks like we'll need a bz for merge, unless
> 
> @adambkaplan - how hard is it to change the jenkins repo to no Feature Freeze ?

@adambkaplan replied in team meeting he will switch this repo to no FF

**Comment by adambkaplan** [comment]
https://github.com/openshift/release/pull/25168 merged, so this can now be merged with the No-FF process.

**Comment by gabemontero** [comment]
/approve

/label px-approved

/label docs-approved

no docs / px required 

@rolfedh FYI ... this is just the QA side, you and I have already merged the doc changes for the actual function here

I'll let @jitendar-singh or @prietyc123 do the qe label

we'll add the looks good to merge label once all the comments have been addressed

**Comment by gabemontero** [comment]
> [openshift/release#25168](https://github.com/openshift/release/pull/25168) merged, so this can now be merged with the No-FF process.

thanks @adambkaplan !!

**Comment by akram** [comment]
This way it looks good to me @jitendar-singh , that looks quite readable and reusable; and I think we could easily extend if for future scenario.
I will /approve it but let @gabemontero lgtm it as he was much more active on the initial reviews

/approve

**Comment by gabemontero** [comment]
/lgtm

I'll leave it to you @jitendar-singh  or @prietyc123 to add the qe approved label

**Comment by prietyc123** [comment]
looks good to me.

/label qe-approved

**Comment by gabemontero** [comment]
the sample operator bump for 4.10 that just merged resulted in nodejs:12 going away since it is EOL

the sync plugin e2e is now hitting that:

`        {err=The ImageStreamTag "nodejs:12" is invalid: from: Error resolving ImageStreamTag nodejs:12 in namespace openshift: unable to find latest tagged image`

I will update the sync plugin e2e's and then we can retest this

**Comment by gabemontero** [comment]
> the sample operator bump for 4.10 that just merged resulted in nodejs:12 going away since it is EOL
> 
> the sync plugin e2e is now hitting that:
> 
> ` {err=The ImageStreamTag "nodejs:12" is invalid: from: Error resolving ImageStreamTag nodejs:12 in namespace openshift: unable to find latest tagged image`
> 
> I will update the sync plugin e2e's and then we can retest this

https://github.com/openshift/jenkins-sync-plugin/pull/413 is up

**Comment by gabemontero** [comment]
will retest in a couple after CI propagates the https://github.com/openshift/jenkins-sync-plugin/pull/413 changes

**Comment by gabemontero** [comment]
install failure on e2e

/retest

**Pull request merged**
